src/main/java/ca/corefacility/bioinformatics/irida/ria/web/analysis/AnalysisController.java

```diff
@@ -151,4 +167,59 @@ public String getAdvancedPhylogeneticVisualizationPage(@PathVariable Long submis
 		model.addAttribute("submission", submission);
 		return BASE + "visualizations/phylocanvas-metadata";
 	}
+
+	/**
+	 * Get the html page from the file name provided.
+	 *
+	 * @param submissionId {@link Long} identifier for an {@link AnalysisSubmission}
+	 * @param filename     The html file name
+	 * @param locale       User's locale
+	 * @param response     {@link HttpServletResponse}
+	 * @throws IOException if we can't write the file to the response
+	 */
+	@RequestMapping("/{submissionId}/html-output")
+	public void getHtmlOutputForSubmission(@PathVariable Long submissionId, @RequestParam String filename,
+			Locale locale, HttpServletResponse response) throws IOException {
+		AnalysisSubmission submission = analysisSubmissionService.read(submissionId);
+		Set<AnalysisOutputFile> files = submission.getAnalysis()
+				.getAnalysisOutputFiles();
+		AnalysisOutputFile outputFile = null;
+		String htmlExt = "html";
+
+		for (AnalysisOutputFile file : files) {
+			if (file.getFile()
+					.toFile()
+					.getName()
+					.contains(filename) && FilenameUtils.getExtension(filename)
+					.equals(htmlExt)) {
+				outputFile = file;
+				break;
+			}
+		}
+
+		// Set the common Http headers
+		response.setHeader(HttpHeaders.CONTENT_DISPOSITION, "inline");
+		response.setHeader(HttpHeaders.CONTENT_TYPE, "text/html");
+
+		try (InputStream inputStream = new FileInputStream(outputFile.getFile()
+				.toString()); OutputStream outputStream = response.getOutputStream()) {
+			response.setHeader(HttpHeaders.CONTENT_LENGTH, String.valueOf(Files.size(outputFile.getFile())));
+			// Copy the file contents to the response outputstream
+			IOUtils.copy(inputStream, outputStream);
+		} catch (IOException e) {
+			logger.debug("Html output not found.");
+			response.setHeader(HttpHeaders.CONTENT_LENGTH, "0");
+			String htmlOutputNotFound = messageSource.getMessage("analysis.html.file.not.found",
+					new Object[] { filename }, locale);
+			OutputStream outputStream = response.getOutputStream();
+			/*
+			Write the htmlNotFound message to the outputstream. We do this
+			so that the page doesn't error and will instead display the
+			message.
+			 */
+			outputStream.write(htmlOutputNotFound.getBytes(StandardCharsets.UTF_8));
+			outputStream.flush();
+			outputStream.close();
```

Comment:
The output stream will get closed anyway with the try-with-resources block right?  Is this needed?

Reply:
Since we reaccess it in the catch block would it still close? Or will it just close the original access stream?

Reply:
Ah I'm sorry, I misread the code.  I think you have it correct.  Since you got a handle on a new stream a couple lines up you'll have to close it.  I thought for some reason that this was the same stream you opened in the try-resources block.  This is good and can stay as-is.